polkadot/xcm/xcm-executor/src/traits/should_execute.rs

```diff
@@ -59,19 +59,36 @@ impl ShouldExecute for Tuple {
 		properties: &mut Properties,
 	) -> Result<(), ProcessMessageError> {
 		for_tuples!( #(
-			match Tuple::should_execute(origin, instructions, max_weight, properties) {
-				Ok(()) => return Ok(()),
-				_ => (),
+			let barrier = core::any::type_name::<Tuple>();
+ 			match Tuple::should_execute(origin, instructions, max_weight, properties) {
+				Ok(()) => {
+					tracing::trace!(
+						target: "xcm::should_execute",
+						?origin,
+						?instructions,
+						?max_weight,
+						?properties,
+						%barrier,
+						"pass barrier",
+					);
+							
+					return Ok(())
+				},
+				Err(error) => {
+					tracing::trace!(
+						target: "xcm::should_execute",
+						?origin,
+						?instructions,
+						?max_weight,
+						?properties,
+						?error,
+						%barrier,
+						"did not pass barrier",
+					);
```

Comment:
I tried using this, but for some reason this never gets printed 🤷‍♂️ too late in the day now to investigate why

Reply:
Hmm.

Should work with https://github.com/paritytech/polkadot-sdk/pull/4863

Reply:
ah, yes, I was missing #4863 on my branch, works now, thanks!